src/CommunityToolkit.Maui/Alerts/Toast/Toast.shared.cs

```diff
@@ -0,0 +1,112 @@
+﻿using CommunityToolkit.Maui.Core;
+using System.ComponentModel;
+
+namespace CommunityToolkit.Maui.Alerts;
+
+/// <inheritdoc/>
+public partial class Toast : IToast
+{
+	/// <inheritdoc/>
+	public string Text { get; set; } = string.Empty;
+
+	/// <inheritdoc/>
+	public ToastDuration Duration { get; set; } = ToastDuration.Short;
+
+	/// <inheritdoc/>
+	public double TextSize { get; set; } = Defaults.FontSize;
+
+	/// <summary>
+	/// Create new Toast
+	/// </summary>
+	/// <param name="message">Toast message</param>
+	/// <param name="duration">Toast duration</param>
+	/// <param name="textSize">Toast font size</param>
+	/// <returns>New instance of Toast</returns>
+	public static IToast Make(
+		string message,
+		ToastDuration duration = ToastDuration.Short,
+		double textSize = Defaults.FontSize)
+	{
+		ArgumentNullException.ThrowIfNull(message);
+
+		if (!Enum.IsDefined(typeof(ToastDuration), duration))
+		{
+			throw new InvalidEnumArgumentException(nameof(duration), (int)duration, typeof(ToastDuration));
+		}
+
+		if (textSize <= 0)
+		{
+			throw new ArgumentOutOfRangeException(nameof(textSize), "Toast font size must be positive");
+		}
+
+		return new Toast()
+		{
+			Text = message,
+			Duration = duration,
+			TextSize = textSize
+		};
+	}
+
+	/// <summary>
+	/// Show Toast
+	/// </summary>
+	public virtual partial Task Show(CancellationToken token = default);
+
+	/// <summary>
+	/// Dismiss Toast
+	/// </summary>
+	public virtual partial Task Dismiss(CancellationToken token = default);
+
+#if !(IOS || ANDROID || MACCATALYST || WINDOWS)
+	/// <summary>
+	/// Show Toast
+	/// </summary>
+	public virtual partial Task Show(CancellationToken token)
+	{
+		token.ThrowIfCancellationRequested();
+		return Task.CompletedTask;
+	}
+
+	/// <summary>
+	/// Dismiss Toast
+	/// </summary>
+	public virtual partial Task Dismiss(CancellationToken token)
+	{
+		token.ThrowIfCancellationRequested();
+		return Task.CompletedTask;
+	}
+#endif
+
+	/// <summary>
+	/// Dispose Toast
+	/// </summary>
+	public async ValueTask DisposeAsync()
+	{
+		await DisposeAsyncCore();
+		GC.SuppressFinalize(this);
+	}
+
+	/// <summary>
+	/// Dispose Toast
+	/// </summary>
+	protected virtual async ValueTask DisposeAsyncCore()
+	{
+#if ANDROID
+		await Device.InvokeOnMainThreadAsync(() => nativeToast?.Dispose());
+#else
+		await Task.CompletedTask;
+#endif
+	}
```

Comment:
Can't we just return the `Task`? That way we don't create the state machine

Reply:
You're totally correct, but I think this is easier to read than the alternative.

The unused state machine has a negligible impact on the DLL size (a couple kilobytes) and a negligible impact on performance (a couple ns).

That being said, I'm fine with either implementation.

### Alternative
```cs
#if ANDROID
	protected virtual async ValueTask DisposeAsyncCore()
	{
		if(NativeToast is not null)
		{
			await Device.InvokeOnMainThreadAsync(() => NativeToast.Dispose());
		}
        }
#else
	protected virtual ValueTask DisposeAsyncCore()
	{
		return new ValueTask();
        }
#endif
```

Reply:
@brminnick I'm wondering if we can remove the `async` modifier for Android implementation as well

protected virtual ValueTask DisposeAsyncCore()
	{
		if(NativeToast is not null)
		{
			return Device.InvokeOnMainThreadAsync(() => NativeToast.Dispose());
		}
        }

Reply:
we cannot: InvokeOnMainThreadAsync returns Task
![image](https://user-images.githubusercontent.com/33021114/152727837-6732c3f5-6f00-4c37-9bf1-509526e83d3e.png)